test/system/groups/samples_test.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'application_system_test_case'
+
+module Groups
+  class SamplesTest < ApplicationSystemTestCase
+    def setup
+      login_as users(:john_doe)
+      @group = groups(:group_one)
+      @namespaces = Namespaces::ProjectNamespace.where(parent_id: @group.self_and_descendant_ids)
+      @samples_count = samples.select { |sample| @namespaces.include?(sample.project.namespace) }.count
+    end
+
+    test 'visiting the index' do
+      visit group_samples_url(@group)
+
+      assert_selector 'h1', text: I18n.t(:'groups.samples.index.title')
+      assert_selector 'tbody > tr', count: @samples_count
```

Comment:
You also might want to add a test to ensure that the paging is working.

Reply:
@ksierks you can take a look at `test/system/projects_test.rb` to see how to test the pagination

Reply:
I thought that's what pagination_component_test.rb is for. Please see 6faf8d2cb905378ccea51ff3b5207f7c67e36e0c.

Reply:
@ksierks The `pagination_component_test.rb` only tests the pagination in isolation. The system testing tests the whole page as a UI/Integration test which we can make sure the correct data is displayed on the subsequent pages